README.md

```diff
@@ -345,27 +345,35 @@ public interface RetryListener {
 
     void open(RetryContext context, RetryCallback<T> callback);
 
+    void onSuccess(RetryContext context, T result);
+
     void onError(RetryContext context, RetryCallback<T> callback, Throwable e);
 
     void close(RetryContext context, RetryCallback<T> callback, Throwable e);
+
 }
 ```
 
-The `open` and `close` callbacks come before and after the entire retry in the simplest
-case, and `onError` applies to the individual `RetryCallback` calls. The close method
-might also receive a `Throwable`. If there has been an error, it is the last one thrown by
-the `RetryCallback`.
+The `open` and `close` callbacks come before and after the entire retry in the simplest case, and `onSuccess`, `onError` apply to the individual `RetryCallback` calls; the current retry count can be obtained from the `RetryContext`.
+The close method might also receive a `Throwable`.
+Starting with version 2.0, the `onSuccess` method is called after a successful call to the callback.
+This allows the listener to examine the result and throw an exception if the result doesn't match some expected criteria.
+The type of the exception thrown is then used to determine whether the call should be retried or not, based on the retry policy.
+If there has been an error, it is the last one thrown by the `RetryCallback`.
 
 Note that when there is more than one listener, they are in a list, so there is an order.
-In this case, `open` is called in the same order, while `onError` and `close` are called
-in reverse order.
+In this case, `open` is called in the same order, while `onSuccess`, `onError`, and `close` are called in reverse order
+
+`RetryListenerSupport` is provided, with no-op implementations; you can extend this class if you don't need to implement all of the `RetryListener` methods.
```

Comment:
I don't think we need this one from now on since we just simply can go ahead with `default` methods in the interface.
Perhaps separate issue though...

Reply:
True, but we'd have to go through a deprecation cycle; definitely a separate issue.